src/main.py

```diff
@@ -104,13 +98,27 @@ def render_page(sensors):
     measure = 'GET /?measure=on' in request
 
     # note that LED is on on low signal
+    measurements = []
     if measure:
         for sensor in sensors:
-            sensor.measure()
+            pass
+            measurements.append(sensor.measure())  # yields a future
+
+    # wait for measurements to actually finish
+    await uasyncio.gather(*measurements)
```

Comment:
Here we now `await` all Futures the `measure` methods have returned to actually complete and return a value. This is done using the `gather` function, as in "gather" results. It takes any number of futures as parameters, so we expand our list `*measurements` into arguments.